src/app/modules/item/pages/chapter-user-progress/chapter-user-progress.component.html

```diff
@@ -47,22 +48,49 @@
           <td *ngFor="let col of columns">
             <ng-container [ngSwitch]="col.field">
               <ng-container *ngSwitchCase="'title'">
-                <a
-                  class="alg-link"
-                  [ngClass]="{'disabled': rowData.id === itemData?.item?.id}"
-                  (click)="onClick(rowData)"
-                >
-                  {{ rowData.title }}
-                </a>
+                <div class="title" *ngIf="itemData && itemData.currentResult?.attemptId as parentAttemptId">
+                  <a
+                    class="alg-link"
+                    [ngClass]="{'disabled': rowData.id === itemData.item.id}"
+                    [routerLink]="rowData | rawItemRoute | withPath : itemData.route.path.concat([ itemData.item.id ]) | withParentAttempt : parentAttemptId | url : [ 'progress', 'chapter-user-progress' ]"
```

Comment:
Using `{type, id} | rawItemRoute` then adding the path, then adding the attempt (attempt or parent attempt)... it looks like we just need a way to build a `itemRoute` or a `fullItemRoute` directly. This way you could use functions which can only be used on  `itemRoute` or `fullItemRoute`. Don't you agree? (I am not 100% sure)

Reply:
Definitely, so `withAnswer`, `withParentAttempt`, `withPath` pipes should be merged with new `itemRoute` pipe?

Reply:
Yes.
Actually, I think "withAnswer" still make sense if you have a route and you want to add an answer (so it is to push a more specific behavior as the "page" does) ... but "withAnswer" does not need to be used here in a case when we build a route from scratch.